warc-indexer/src/main/java/uk/bl/wa/annotation/Annotator.java

```diff
@@ -177,6 +179,20 @@ public void applyAnnotations(URI uri, SolrInputDocument solr)
 				updateCollections(subdomains.get(key), solr, crawl_dates);
 			}
 		}
+		// "All source_file that match this source_file_prefix"
+		Pattern pattern;
+		Matcher matcher;
+		String sourceFile = (String) solr.getField(SolrFields.SOURCE_FILE).getValue();
+		HashMap<String, UriCollection> sourceFilePrefix = this.annotations
+				.getCollections().get("source_file_prefix");
+		for (String key : sourceFilePrefix.keySet()) {
+			LOG.debug("Applying source_file_prefix annotations for: " + key);
+			pattern = Pattern.compile(key);
```

Comment:
This all looks good, the only thing is the parameter is called `source_file_prefix` but you're actually using a full regular expression. This seems like a good idea, but in that case we should probably call is `source_file_matches`.